File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Node.java
package com.example.grts;

import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.GraphQLInterface;
import viaduct.java.api.types.NodeCompositeOutput;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.List;

public interface Node extends NodeCompositeOutput {

    Type<Node> Reflection = Type.ofClass(Node.class);

    final class Fields implements TypeFields<Node> {
        private Fields() {}

        public static final Field<Node> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Node> id =
                                Field.of("id", Reflection);

    }

    GlobalID<? extends Node> getIdOrThrow(String alias);
    GlobalID<? extends Node> getIdOrThrow();
    GlobalID<? extends Node> getId(String alias);
    GlobalID<? extends Node> getId();

}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Node.kt
@file:Suppress("warnings")

package viaduct.api.grts

interface Node : viaduct.api.types.Interface, viaduct.api.types.NodeCompositeOutput {
          fun getIdOrThrow(alias: String?): viaduct.api.globalid.GlobalID<viaduct.api.grts.Node>
          fun getIdOrThrow(): viaduct.api.globalid.GlobalID<viaduct.api.grts.Node>
          fun getId(alias: String?): viaduct.api.globalid.GlobalID<viaduct.api.grts.Node>?
          fun getId(): viaduct.api.globalid.GlobalID<viaduct.api.grts.Node>?


    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Node> {
        override final val name = "Node"
        override final val kcls = viaduct.api.grts.Node::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Node> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Node> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Node.Reflection)

            final val id: viaduct.api.reflect.Field<viaduct.api.grts.Node> =
                viaduct.api.internal.FieldImpl("id", viaduct.api.grts.Node.Reflection)

    }

}